app/src/main/java/io/gnosis/kouban/ui/balances/BalancesWidgetPrefs.kt

```diff
@@ -0,0 +1,34 @@
+package io.gnosis.kouban.ui.balances
+
+import android.content.Context
+import android.content.SharedPreferences
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.common.utils.edit
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.asEthereumAddressString
+
+class BalancesWidgetPrefs(context: Context) {
+
+    private val prefs: SharedPreferences
+
+    init {
+        prefs = context.getSharedPreferences(PREFS_WIDGET_BALANCES, Context.MODE_PRIVATE)
+    }
+
+    fun saveTokenForWidget(token: Solidity.Address, appWidgetId: Int) {
+        prefs.edit {
+            putString(KEY_WIDGET_TOKEN, "${appWidgetId}.${token.asEthereumAddressString()}")
+        }
+    }
+
+    fun loadTokenForWidget(appWidgetId: Int): Solidity.Address? {
+        return prefs.getString(KEY_WIDGET_TOKEN, null)?.split(".")?.get(1)?.asEthereumAddress()
```

Comment:
We never use the widget id and it adds quite a lot of messy code (split etc) can we remove it?

Reply:
How so, you only use one prefs key

Reply:
ups... Sorry, I thought I have implemented it. Now each widget setting is saved using 2 keys  ```private val KEY_WIDGET_TOKEN_ = "key_widget_token_"
        private val KEY_WIDGET_ID_ = "key_widget_id_"```